### DISCUSSION TITLE
Flutter cannot send api request

### DISCUSSION THREAD

**Discussion by itsatifsiddiqui** [deepgram/community#622] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

I am trying to post a request using dio package in flutter, the post request is working fine in postman and i am using the code snipped of dart Dio from postman to flutter but its not working in the app, its giving me 400 error.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?detect_entities=true&punctuate=true

### If possible, please attach your code or paste it into the text box.

 final audioFile =
                  await File(transcription.audioFilePath!).create();
              audioFile.log();
              'Started'.log();
              var headers = {
                'Authorization':
                    'Token $apiKey'
              };
              var data = FormData.fromMap({
                'files': [
                  await MultipartFile.fromFile(
                    audioFile.path,
                    filename: transcription.audioFilePath!.split('/').last,
                  )
                ],
              });

              var dio = Dio();
              var response = await dio.request(
                'https://api.deepgram.com/v1/listen?detect_entities=true&punctuate=true',
                options: Options(
                  method: 'POST',
                  headers: headers,
                ),
                data: data,
                onSendProgress: (count, total) {
                  count.log();
                  total.log();
                },
              );

              if (response.statusCode == 200) {
                print(json.encode(response.data));
              } else {
                print(response.statusMessage);
              }

              'Ended'.log();

### If possible, please attach an example audio file to reproduce the issue.

https://github.com/deepgram/community/assets/46827004/d6fb347f-1bd5-4319-8101-6e9e3c85a54f

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
@itsatifsiddiqui  you might want to check out this blog on using[ Deepgram with Flutter](https://deepgram.com/learn/flutter-speech-to-text-tutorial) to help you troubleshoot your code.  I also found this [project on GitHub ](https://github.com/acastry/deepgram-live-transcription-flutter)which might help you.
